services_app/src/androidTest/java/org/opendatakit/webkitserver/service/OdkWebserverServiceTest.java

```diff
@@ -185,69 +185,130 @@ public void testBindingNRestart() {
         }
     }
 
+
     @Test
     public void testServingHelloWorldHtml() {
-        ODKFileUtils.verifyExternalStorageAvailability();
-        ODKFileUtils.assertDirectoryStructure(TestConsts.APPNAME);
-
+        // Arrange
+        File directoryLocation = createTestDirectory();
+        File fileLocation = createTestFile(directoryLocation);
+        // Act
+        IWebkitServerInterface serviceInterface = getIWebkitServerInterface();
+        restartService(serviceInterface);
+        // Assert
+        assertResponseMatchesHelloWorldHtml(fileLocation);
+    }
+    private File createTestDirectory() {
         File directoryLocation = new File(ODKFileUtils.getConfigFolder(TestConsts.APPNAME), TEST_DIR);
-        File fileLocation = new File(directoryLocation, TEST_FILE_NAME);
+        if (!directoryLocation.isDirectory()) {
+            directoryLocation.mkdirs();
+        }
+        return directoryLocation;
+    }
 
-        IWebkitServerInterface serviceInterface = getIWebkitServerInterface();
+    private File createTestFile(File directoryLocation) {
+        File fileLocation = new File(directoryLocation, TEST_FILE_NAME);
 
-        PrintWriter writer = null;
-        try {
-            if(!directoryLocation.isDirectory()) {
-                directoryLocation.mkdirs();
-            }
-            writer = new PrintWriter(fileLocation, "UTF-8");
+        try (PrintWriter writer = new PrintWriter(fileLocation, "UTF-8")) {
```

Comment:
We found some mobile devices did not write the file out without a flush (it delays the writing of files for optimizations) therefore, to resolve transient issues we needed to use flush to make sure the OS completes the operation we are testing before moving on.